graphql_compiler/deserialization.py

```diff
@@ -0,0 +1,174 @@
+    if value in true_values:
+        return True
+    elif value in false_values:
+        return False
+    else:
+        raise ValueError(
+            f"Received unexpected GraphQLBoolean value {value} of type {type(value)}. Expected one "
+            f"of: {true_values + false_values}."
+        )
+
+
+_CUSTOM_SCALAR_DESERIALIZATION_FUNCTIONS: Mapping[str, Callable[[Any], Any]] = MappingProxyType(
+    {
+        # Bypass the GraphQLFloat parser and allow strings as input. The JSON spec allows only
+        # for 64-bit floating point numbers, so large floats might have to be represented as
+        # strings.
+        GraphQLFloat.name: float,
+        # Bypass the GraphQLInt parser and allow long ints and strings as input. The JSON spec
+        # allows only for 64-bit floating point numbers, so large ints might have to be
+        # represented as strings.
+        GraphQLInt.name: int,
+        # Bypass the GraphQLBoolean parser and allow some strings and ints as input.
+        GraphQLBoolean.name: _custom_boolean_deserialization,
+    }
+)
+
+_JSON_TYPES_AND_DESERIALIZATION_FUNCTIONS: Mapping[
+    str, Tuple[Tuple[Type, ...], Callable[[Any], Any]]
+] = MappingProxyType(
+    {
+        scalar_type.name: (
+            _ALLOWED_SCALAR_TYPES[scalar_type.name],
+            _CUSTOM_SCALAR_DESERIALIZATION_FUNCTIONS.get(scalar_type.name, scalar_type.parse_value),
+        )
+        for scalar_type in SUPPORTED_SCALAR_TYPES
+    }
+)
+
+
+def deserialize_scalar_value(expected_type: GraphQLScalarType, value: Any) -> Any:
+    """Convert a scalar value to the appropriate type for the given GraphQLScalarType.
+
+    Below are examples of accepted encodings of all the types:
+        GraphQLDate: "2018-02-01"
+        GraphQLDateTime: "2018-02-01T05:11:54Z"
+        GraphQLFloat: 4.3, "5.0", 5
+        GraphQLDecimal: "5.00000000000000000000000000001"
+        GraphQLInt: 4, "3803330000000000000000000000000000000000000000000"
+        GraphQLString: "Hello"
+        GraphQLBoolean: True, 1, "1", "True", "true"
+        GraphQLID: "13d72846-1777-6c3a-5743-5d9ced3032ed"
+
+    Args:
+        expected_type: a GraphQLScalarType to which value should be converted.
+        value: object that can be interpreted as being of expected_type.
+
+    Returns:
+        a value of the type produced by the parser of the expected type:
+            GraphQLDate: datetime.date
+            GraphQLDateTime: datetime.datetime with tzinfo=pytz.utc
+            GraphQLFloat: float
+            GraphQLDecimal: decimal.Decimal
+            GraphQLInt: int
+            GraphQLString: str
+            GraphQLBoolean: bool
+            GraphQLID: str
+
+    Raises:
+        AssertionError: if the expected_type does not have a known deserialization function.
+        ValueError: if the value is not appropriate for the type. ValueError is chosen because
+            it is already the base case of exceptions raised by the GraphQL parsers.
```

Comment:
nit: Can you fix the indentation here so that "it" is aligned with "if"? Sorry, I think that this indentation issue comes from the code I attached to the comment I made earlier. Github code formatting was being kind of weird.